### DISCUSSION TITLE
How to load Avalonia Application to system tray

### DISCUSSION THREAD

**Discussion by Aswathips** [AvaloniaUI/Avalonia#8241] [Q&A] [answered]
Hi,
How can we load an Avalonia Application to system tray with menu items.  Also I want to set a system tray icon while loading the app to system tray and I need to change that icon after some time.
Anyone please explain me this with a simple example.

**Comment by maxkatz6** [comment, marked as answer]
You can add a tray icon with a menu:
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/App.xaml#L34-L53

**Reply by Aswathips** [reply]
Actually Iam working on Avalonia application not on Avalonia MVVM application.  So how to use this code in Avalonia application 
and also I don't want to show the app in system tray at the start of the app.

**Reply by maxkatz6** [reply]
> Avalonia MVVM application

It doesn't matter. NativeMenuItem has Click event

> and also I don't want to show the app in system tray at the start of the app.

You can set value to this TrayIcons property at runtime when you need it.

**Reply by Aswathips** [reply]
how can I set value to this TrayIcons property at runtime.
please give me a example

**Reply by maxkatz6** [reply]
```TrayIcon.SetIcons(Application.Current, new TrayIcons());```
or
```var iconsList = TrayIcon.GetIcons(Application.Current);```

**Reply by Aswathips** [reply]
Using this I get the properties of the TrayIcons. 
But while I try to change the Tray Icon there occur the error

Cannot implicitly convert type 'string' to 'Avalonia.Controls.WindowIcon'

**Reply by maxkatz6** [reply]
Check IDE API reference, types documentation or the source code. 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/WindowIcon.cs#L17

**Reply by Aswathips** [reply]
how can I set value to this TrayIcons NativeMenuItem property at runtime.
Actually I want to change the NativeMenuItem Header value at runtime.
